apps/next/pages/emgg.tsx

```diff
@@ -0,0 +1,444 @@
+                    banner,
+                },
+            };
+        } else {
+            // if they have their twitch account connected, but dont have a banner yet
+            if (session.accounts['twitch']) {
+                // fetch twitch username, and save a banner for them (this code should run only once for every user after they sign up)
+                const usernameInfo = await localAxios.get(`/api/twitch/username/${session.userId as string}`);
+                const username = usernameInfo.data.displayName;
+                return {
+                    props: {
+                        banner: {
+                            ...defaultBannerSettings,
+                            foregroundProps: {
+                                ...defaultBannerSettings.foregroundProps,
+                                username,
+                            },
+                        },
+                    },
+                };
+            } else {
+                // if they need to still connect their twitch account
+                return {
+                    props: {
+                        banner: {},
+                    },
+                };
+            }
+        }
+    }
+    // if they are not logged in
+    return {
+        props: {
+            banner: {},
+        },
+    };
+};
+
+export default function Page({ banner }: Props) {
+    const { data: sessionInfo } = useSession();
+
+    const { data: paymentPlanResponse } = useSWR<APIPaymentObject>('payment', async () => (await fetch('/api/user/subscription')).json());
+    const paymentPlan: PaymentPlan = paymentPlanResponse === undefined ? 'Free' : paymentPlanResponse.plan;
+    const { colorMode, toggleColorMode } = useColorMode();
+
+    useEffect(() => {
+        if (colorMode === 'dark') return;
+        toggleColorMode();
+    });
+    const { data: streamingState } = useSWR('streamState', async () => await (await fetch(`/api/twitch/streaming/${session?.user['id']}`)).json());
+    const streaming = streamingState ? streamingState.isStreaming : false;
+
+    const bgId = 'ImageBackground';
+    const fgId = 'Emgg';
+    const bgProps = {
+        src: 'https://cdn.discordapp.com/attachments/922692527625220126/932410278132477972/emgg.png',
+    };
+    const [fgProps, setFgProps] = useState(banner.foregroundProps ?? (ForegroundTemplates[defaultForeground].defaultProps as any));
+    const [reAuth, setReAuth] = useState(false);
+
+    const BackgroundTemplate = BackgroundTemplates[bgId];
+    const ForegroundTemplate = ForegroundTemplates[fgId];
+
+    const toast = useToast();
+    const breakpoint = useBreakpoint();
+
+    const { ensureSignUp, isOpen, onClose, session } = useConnectToTwitch();
+
+    const styles: BoxProps = {
+        background: 'whiteAlpha.100',
+    };
+
+    const [isToggling, { on, off }] = useBoolean(false);
+
+    useEffect(() => {
+        axios.post('/').then(() => {
+            console.log('attempt to clear 301 redirect from cache');
```

Comment:
remove this or change to datadog

Reply:
Actually we don't need this thing at all

Reply:
Yeah, probably can remove it now